### PR TITLE
feat(query): copy with transform support max_files

### LINKED ISSUE #10803: Feature: copy with transform support max_files too
**Summary**

update bind_copy_from_query_into_table

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #10803

### CODE REVIEW

File: src/common/storage/src/stage.rs [hunk 1]
```diff
@@ -189,11 +192,16 @@ impl StageFilesInfo {
                 if first_only {
                     break;
                 }
+                limit += 1;
+                if limit == max_files {
+                    println!("in blocking_list, res is {:?}", res.clone());
```
Comment (youngsofun) [inline src/common/storage/src/stage.rs:197]:
```suggestion
```

File: src/query/storages/parquet/src/parquet_table/partition.rs [hunk 2]
```diff
@@ -114,7 +114,7 @@ impl ParquetTable {
                 .map(|f| (f.path.clone(), f.size))
                 .collect::<Vec<_>>(),
             None => if self.operator.info().can_blocking() {
-                self.files_info.blocking_list(&self.operator, false)
+                self.files_info.blocking_list(&self.operator, false, None)
```
Comment (youngsofun) [inline src/query/storages/parquet/src/parquet_table/partition.rs:117]: I can't quite remember it:  when will the code switch to the "None=>" arm (maybe unreachable)?  
to make sure is ok to omit max_file here

File: tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.result [hunk 3]
```diff
@@ -0,0 +1,35 @@
+6
```
Comment (youngsofun) [inline tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.result:1]: why  "6"  in the first line?

File: tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+echo "drop stage if exists s5;" | $MYSQL_CLIENT_CONNECT
+echo "create stage s5 url = 'fs:///tmp/00_0005/data/' FILE_FORMAT = (type = PARQUET)" | $MYSQL_CLIENT_CONNECT
+
+
+for force in 'false'  'true'
+do
+	for purge in 'false'  'true'
+	do
+	  gen_files
+	  echo "select count(*) from @s5" | $MYSQL_CLIENT_CONNECT
```
Comment (youngsofun) [inline tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh:36]: is " echo "select count(*) from @s5" | $MYSQL_CLIENT_CONNECT" on purpose of for debug?

Comment (TCeason) [reply tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh:36]: Yes. And I think we can save it.

Comment (youngsofun) [reply tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh:36]: if you really want to add it,  better put it after `echo "--- force = ${force}, purge = ${purge}"`?

File: src/query/storages/parquet/src/parquet_table/partition.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/storages/parquet/src/parquet_table/partition.rs:117]: Oh, I recall now， select stage （no copy ）

File: tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh (hunk 4 above)
Comment (youngsofun) [reply tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh:36]: which make results more readable.

**Comment by youngsofun** [comment]
rest lgtm

File: tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh (hunk 4 above)
Comment (TCeason) [reply tests/suites/1_stateful/00_copy/00_0005_copy_from_stage_max_files.sh:36]: ok

**Review by youngsofun — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**